File: src/main/java/model/UserAddress.java
package model;

import jakarta.persistence.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.Nationalized;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "UserAddresses")
public class UserAddress {
    @Id
    @ColumnDefault("newid()")
    @Column(name = "AddressID", nullable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "UserID", nullable = false)
    private User userID;

    @Nationalized
    @Column(name = "AddressType", nullable = false, length = 20)
    private String addressType;

    @Nationalized
    @Column(name = "AddressLine1", nullable = false)
    private String addressLine1;

    @Nationalized
    @Column(name = "AddressLine2")
    private String addressLine2;

    @Nationalized
    @Column(name = "City", nullable = false, length = 100)
    private String city;

    @Nationalized
    @Column(name = "State", nullable = false, length = 100)
    private String state;

    @Nationalized
    @Column(name = "PostalCode", nullable = false, length = 20)
    private String postalCode;

    @Nationalized
    @Column(name = "Country", nullable = false, length = 100)
    private String country;

    @ColumnDefault("0")
    @Column(name = "IsDefault")
    private Boolean isDefault;

    @ColumnDefault("getdate()")
    @Column(name = "CreatedDate")
    private Instant createdDate;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public User getUserID() {
        return userID;
    }

    public void setUserID(User userID) {
        this.userID = userID;
    }

    public String getAddressType() {
        return addressType;
    }

    public void setAddressType(String addressType) {
        this.addressType = addressType;
    }

    public String getAddressLine1() {
        return addressLine1;
    }

    public void setAddressLine1(String addressLine1) {
        this.addressLine1 = addressLine1;
    }

    public String getAddressLine2() {
        return addressLine2;
    }

    public void setAddressLine2(String addressLine2) {
        this.addressLine2 = addressLine2;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getPostalCode() {
        return postalCode;
    }

    public void setPostalCode(String postalCode) {
        this.postalCode = postalCode;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public Boolean getIsDefault() {
        return isDefault;
    }

    public void setIsDefault(Boolean isDefault) {
        this.isDefault = isDefault;
    }

    public Instant getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Instant createdDate) {
        this.createdDate = createdDate;
    }

}

File: src/main/java/model/UserAddress.kt
package model

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import org.hibernate.annotations.Nationalized
import java.time.Instant
import java.util.*

@Entity
@Table(name = "UserAddresses")
open class UserAddress {
    @Id
    @ColumnDefault("newid()")
    @Column(name = "AddressID", nullable = false)
    open var id: UUID? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "UserID", nullable = false)
    open var userID: User? = null

    @Nationalized
    @Column(name = "AddressType", nullable = false, length = 20)
    open var addressType: String? = null

    @Nationalized
    @Column(name = "AddressLine1", nullable = false)
    open var addressLine1: String? = null

    @Nationalized
    @Column(name = "AddressLine2")
    open var addressLine2: String? = null

    @Nationalized
    @Column(name = "City", nullable = false, length = 100)
    open var city: String? = null

    @Nationalized
    @Column(name = "State", nullable = false, length = 100)
    open var state: String? = null

    @Nationalized
    @Column(name = "PostalCode", nullable = false, length = 20)
    open var postalCode: String? = null

    @Nationalized
    @Column(name = "Country", nullable = false, length = 100)
    open var country: String? = null

    @ColumnDefault("0")
    @Column(name = "IsDefault")
    open var isDefault: Boolean? = null

    @ColumnDefault("getdate()")
    @Column(name = "CreatedDate")
    open var createdDate: Instant? = null
}